src/Workspaces/MSBuildWorkspaceLoader.cs

```diff
@@ -29,35 +31,45 @@ internal static class MSBuildWorkspaceLoader
                 { "AlwaysCompileMarkupFilesInSeparateDomain", bool.FalseString },
             };
 
-            var workspace = MSBuildWorkspace.Create(properties);
+            MSBuildWorkspace workspace;
 
-            Build.Framework.ILogger? binlog = null;
-            if (createBinaryLog)
+            await s_guard.WaitAsync();
```

Comment:
This is needed because tests run in parallel but msbuild won't

Reply:
Still don't quite understand this. Why can't each test create their own workspace?

Reply:
Since the CodeFormatterTests is creating MSBuildWorkspaces and now we are creating a MSBuildWorkspace here, there is an InvalidOperationException "Cannot start a new build while a build is in progress"

Reply:
Could we create a `WorspaceFact` that controls this intead of changing product code?

Reply:
Took a stab at this in https://github.com/dotnet/format/pull/899